jetbrains-core/src/software/aws/toolkits/jetbrains/core/toolwindow/ToolkitToolWindow.kt

```diff
@@ -4,135 +4,118 @@
 package software.aws.toolkits.jetbrains.core.toolwindow
 
 import com.intellij.openapi.Disposable
-import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.application.runInEdt
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.wm.RegisterToolWindowTask
 import com.intellij.openapi.wm.ToolWindow
-import com.intellij.openapi.wm.ToolWindowAnchor
 import com.intellij.openapi.wm.ToolWindowManager
 import com.intellij.ui.content.Content
-import com.intellij.ui.content.impl.ContentImpl
-import icons.AwsIcons
-import java.util.concurrent.ConcurrentHashMap
-import javax.swing.Icon
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
 import javax.swing.JComponent
 
 interface ToolkitToolWindow {
+    val toolWindow: ToolWindow
+
     fun addTab(
         title: String,
         component: JComponent,
         activate: Boolean = false,
         id: String = title,
-        disposable: Disposable? = null,
+        disposeLater: Disposable? = null,
         refresh: () -> Unit = { }
```

Comment:
Lets kill auto-refresh, each table in CW (only place this i used?) already have refresh buttons on them